UnitsNet/CustomCode/UnitParser.cs

```diff
@@ -54,13 +54,14 @@ internal static class UnitParser
 
             const string exponentialRegex = @"(?:[eE][-+]?\d+)?)";
 
-            string regexString = string.Format(@"(?:\s*(?<value>[-+]?{0}{1}{2}{3})?{4}{5}",
+            string regexString = string.Format(@"(?:\s*(?<value>[-+]?{0}{1}{2}{3})?",
                 numRegex, // capture base (integral) Quantity value
                 exponentialRegex, // capture exponential (if any), end of Quantity capturing
                 @"\s?", // ignore whitespace (allows both "1kg", "1 kg")
-                @"(?<unit>[^\s\d,]+)", // capture Unit (non-whitespace) input
-                @"(and)?,?", // allow "and" & "," separators between quantities
-                @"(?<invalid>[a-z]*)?"); // capture invalid input
+                @"(?<unit>[^\d]+)"); // capture Unit (non-numeric)
```

Comment:
This works with our current cases, but we might stumble onto problems here later if we get abbreviations with numbers in them. As an example, we currently only list `m²` in `Area.Squaremeter`, but for convenience sake we might want to also add `m^2` so that they can parse both variants. The latter would not work with this implementation.
I'm fine with postponing that until we cross that bridge, but one option is to define the regex as a number immediately followed by any of the known unit abbreviations for the specified culture (`IFormatProvider`).

Something like this should work:
```
string valueRegex = $"{numRegex}{exponentialRegex}";
string unitRegex = "(m²|m\^2|cm²|cm\^2)";
string valueUnitRegex = $"(?<value>{valueRegex})\s?(?<unit>{unitRegex})"`;
```

The `unitRegex` string values can be enumerated by something like this:
```
var unitSystem = UnitSystem.GetCached(formatProvider);
string[] abbrevsForAllUnits = 
    Enum.GetValues(typeof(TUnit))
    .Cast<TUnit>()
    .SelectMany(unitSystem.GetAllAbbreviations)
    .Select(Regex.Escape)
    .ToArray();
```

And you'd need to refactor the `UnitParser.ParseUnit()` method a bit:
* Add `where TUnit : struct, IComparable, IFormattable` (should have been added to begin with)